libs/payments/ui/src/lib/client/components/CheckoutCheckbox.tsx

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+'use client';
+
+import * as HoverCard from '@radix-ui/react-hover-card';
+import { useState } from 'react';
+import { Localized } from '@fluent/react';
+
+interface CheckoutCheckboxProps {
+  isRequired: boolean;
+  termsOfService: string;
+  privacyNotice: string;
+  notifyCheckboxChange: (isChecked: boolean) => void;
+}
+
+export function CheckoutCheckbox({
+  isRequired,
+  termsOfService,
+  privacyNotice,
+  notifyCheckboxChange,
+}: CheckoutCheckboxProps) {
+  const [isChecked, setIsChecked] = useState(false);
+
+  const changeHandler = () => {
+    const newValue = !isChecked;
+    setIsChecked(newValue);
+    notifyCheckboxChange(newValue);
+  };
+
+  return (
+    <div className="mt-6">
```

Comment:
Nit: Could we just have `HoverCard.Root` as the parent and update `label`'s class to `flex gap-5 items-center mt-6` to get rid of the nested div?